File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int left=0;
        int right=1;
        int maxp=0;
        while(left<right && right<prices.length){
            if(prices[left]<prices[right]){
                maxp=Math.max(maxp,prices[right]-prices[left]);
            }
            else{
                left=right;
            }
            right++;
        }
        return maxp;
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def dfs(curr, max_so_far):
            if not curr:
                return 0
                
            my_score = 1 if curr.val >= max_so_far else 0
            new_max = max(max_so_far, curr.val)
            
            
            left_score = dfs(curr.left, new_max)
            right_score = dfs(curr.right, new_max)
            
         
            return my_score + left_score + right_score

        return dfs(root, root.val)